src/cdk-experimental/scrolling/virtual-scroll-viewport.ts

```diff
@@ -122,6 +128,9 @@ export class CdkVirtualScrollViewport implements OnInit, OnDestroy {
       this._ngZone.run(() => {
         this._renderedRangeSubject.next(this._renderedRange = range);
         this._changeDetectorRef.markForCheck();
+        this._ngZone.runOutsideAngular(() => this._ngZone.onStable.pipe(take(1)).subscribe(() => {
```

Comment:
It seems a little weird to run outside Angular and then waiting for the zone to stabilize. What about flipping them around like:

```
ngZone.onStable.pipe(take(1)).subscribe(() => {
  ngZone.runOutsideAngular(() => onContentCentered());
});
```

Reply:
Wouldn't subscribing within the zone cause another change detection when the zone stabilizes?

Reply:
Does NgZone patch into the observable subscriptions? AFAIK it only does browser APIs.

Reply:
@jelbourn I was worried about that too, but I tested by creating a component that did:

```ts
ngAfterContentChecked() {
  this.ngZone.onStable.subscribe(() => {})
}
```

It didn't go into a death spiral, so it looks like @crisbeto's suggestion would work.

To me it seems clearer to have it the way it is though. It makes it obvious that I don't want any change detection as a result of this subscription.